### ISSUE TITLE
Schema-driven Validation

### ISSUE THREAD

**Issue by harshit-paneri** [layer5io/layer5#6005]
### Implementation

*  Schema Design for API Validation
*  Ensure that the Golang handlers of the Meshery Server REST APIs uniformly and programmatically (in a schema-driven way) automatically validate requests that they receive against the endpoints’ respective schema. 
* [Meshery API Endpoints](https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399)
### Acceptance Tests

@leecalcote Can review the proposal.

---
<img src="https://raw.githubusercontent.com/layer5io/layer5/master/.github/assets/images/layer5/5-light-small.svg" width="24px" align="left" /><h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
-  🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.layer5.io).

<img src="https://raw.githubusercontent.com/layer5io/layer5/master/.github/assets/images/buttons/community.webp" height="22px" align="left" />Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

**Comment by leecalcote** [comment]
Thank, @harshit-paneri!

**Comment by makifyegin** [comment]
Hi @leecalcote  @harshit-paneri can i work on this ?

**Comment by leecalcote** [comment]
@n2h9...

**Comment by espirado** [comment]
@leecalcote is this issue still open to be taken ?

**Comment by leecalcote** [comment]
@espirado, yes, it is. We have yet to implement a consistent approach to schema validation in https://github.com/meshery, which is where this issue should be opened (not under https://github.com/layer5io).
